test/cpp/jit/test_gpu.cpp

```diff
@@ -7615,6 +7609,96 @@ TEST(NVFuserTest, FusionGroupGuardRelaxedCheck_CUDA) {
   TORCH_CHECK(complyWith(t1, tensor_type));
 }
 
+TEST(NVFuserTest, FusionDisjointSet_CUDA) {
```

Comment:
@shmsong I added a unit test for DisjointSet. Please add more tests if you have some other use cases that should be tested.